includes/abstracts/llms-abstract-user-engagement.php

```diff
@@ -159,4 +218,83 @@ public function is_awarded() {
 
 	}
 
+	/**
+	 * Allow child classes to merge the post content based on content from the template.
+	 *
+	 * @since [version]
+	 *
+	 * @return string
+	 */
+	public function merge_content() {
+		return $this->get( 'content', true );
+	}
+
+	/**
+	 * Update the earned engagement by regenerating it from its template.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $context Sync context. Either "update" for an update to an existing earned engagement
+	 *                        or "create" when the earned engagement is being created.
+	 * @return boolean Returns false if the parent doesn't exist, otherwise returns true.
+	 */
+	public function sync( $context = 'update' ) {
+
+		$template_id = $this->get( 'parent' );
+		$template    = $this->get_template_object( $template_id, true );
+		if ( ! $template ) {
+			return false;
+		}
+
+		$this->set( 'title', get_post_meta( $template_id, "_llms_{$this->model_post_type}_title", true ) );
+		if ( get_post_thumbnail_id( $template_id ) !== get_post_thumbnail_id( $this->get( 'post' ) ) &&
+			! set_post_thumbnail( $this->get( 'post' ), get_post_thumbnail_id( $template_id ) )
+		) {
+			delete_post_thumbnail( $this->get( 'post' ) );
+		}
+
+		$props = array(
+			'content',
+		);
+
+		// If using the block editor, also sync all layout properties.
+		if ( 2 === $template->get_template_version() ) {
+			$props = array_merge(
+				$props,
+				array(
+					'background',
+					'height',
+					'margins',
+					'orientation',
+					'size',
+					'unit',
+					'width',
+				)
+			);
+		}
+
+		foreach ( $props as $prop ) {
+			$raw = 'content' === $prop;
+			$this->set( $prop, $template->get( $prop, $raw ) );
+		}
```

Comment:
All of these properties only apply to certificates

I think this method should be overloaded for certificates, call the parent method, and then run this bit of code to sync this additional meta.

Reply:
Should the overriding method also call the `llms_{$this->model_post_type}_synchronized` action hook?

Reply:
Wouldn't that mean that the base method sets the content and then the overriding method sets it again (with merged data). That would be two database writes instead of one.

Reply:
mmm i didn't consider that... Ideally we'd want that called last.

Alright what do you think about a stub function in the abstract, something like `sync_meta()` which is responsible for performing additional metadata sync. The stub does nothing but extending classes (like certificates in this case) can override it. We can pass the existing instantiated `$template` to that class so it's available without repeating the lookup.

How's that sound?

Reply:
I like the sound of the stub method.